src/sys/socket/addr.rs

```diff
@@ -574,28 +603,31 @@ impl UnixAddr {
                                      ret.sun_path.as_mut_ptr().offset(1) as *mut u8,
                                      path.len());
 
-            Ok(UnixAddr(ret, path.len() + 1))
+            Ok(UnixAddr::from_raw_parts(ret, path.len() + 1))
         }
     }
 
-    fn sun_path(&self) -> &[u8] {
-        unsafe { slice::from_raw_parts(self.0.sun_path.as_ptr() as *const u8, self.1) }
+    /// Create a UnixAddr from a raw `sockaddr_un` struct and a size.
+    ///
+    /// # Safety
+    /// This pair of sockaddr_un+size must be a valid unix addr, which means:
+    /// - sun_len <= sockaddr_un.sun_path.len()
+    /// - if this is a unix addr with a fs path, sun.sun_path is a nul-terminated fs path and
+    ///   sun.sun_path[sun_len - 1] == 0
+    pub unsafe fn from_raw_parts(sun: libc::sockaddr_un, sun_len: usize) -> UnixAddr {
```

Comment:
Why `pub`?

Reply:
I didn't want to remove the functionality that was there previously, creating a `UnixAddr` from a raw `sockaddr_un + usize`, but it should be unsafe now.

Reply:
There probably wasn't a good reason for anybody to manually construct a `UnixAddr` before.  Let's mark this new constructor as `deprecated`.